p2p/ipld/nmt_wrapper.go

```diff
@@ -52,11 +57,14 @@ func (w *ErasuredNamespacedMerkleTree) Push(data []byte, idx rsmt2d.SquareIndex)
 		panic("pushed past predetermined square size")
 	}
 
-	// use the parity namespace if the cell is not in Q0 of the extended datasquare
-	// if the cell is empty it means we got an empty block so we need to use TailPaddingNamespaceID
+	// use the parity namespace if the cell is not in Q0 of the extended
+	// datasquare if the cell is empty it means we got an empty block so we need
+	// to use TailPaddingNamespaceID
 	if idx.Axis+1 > uint(w.squareSize) || idx.Cell+1 > uint(w.squareSize) {
 		copy(nsID, types.ParitySharesNamespaceID)
 	} else {
+		// empty shares use the TailPaddingNamespaceID if the data is empty, so
+		// here we check if the share is empty (namepsace == [0,0,0,0,0,0,0,0])
 		if bytes.Equal(data[:types.NamespaceSize], nsID) {
```

Comment:
What would help with understanding the code better, is to introduce another variable (if there isn't a package constant already) that was called `emptyNsID`, then the if-condition would read:
```suggestion
		// empty shares use the TailPaddingNamespaceID if the data is empty
		if bytes.Equal(data[:types.NamespaceSize], emptyNsID) {
```

Reply:
:100: https://github.com/lazyledger/lazyledger-core/pull/306/commits/422c5732a87792984e15b21a239fcf0da7bad974